lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1040,10 +1067,9 @@ class HSAQueue final : public KalmarQueue
 
 
     void waitForStreamDeps (KalmarAsyncOp *newOp) {
-        hc::memory_scope releaseScope;
-        std::shared_ptr<KalmarAsyncOp> depOp = detectStreamDeps(newOp->getCommandKind(), newOp, &releaseScope);
+        std::shared_ptr<KalmarAsyncOp> depOp = detectStreamDeps(newOp->getCommandKind(), newOp);
         if (depOp != nullptr) {
-            EnqueueMarkerWithDependency(1, &depOp, HCC_OPT_FLUSH ? releaseScope : hc::system_scope);
+            EnqueueMarkerWithDependency(1, &depOp, HCC_OPT_FLUSH ? hc::no_scope : hc::system_scope);
```

Comment:
Correct, this is a key part of the change.   waitForStreamDeps is the HCC-internal function used to wait for commands in the same stream., specifically used for kernels or markers that need to wait for copy operations.    With HCC_OPT_FLUSH=1 now tracking and providing flushes in the appropriate places through other mechanisms, this marker is only needed for ordering and not cache flushing and thus can use the faster release synchronization.